### ISSUE TITLE
Refactor tests to cover both global and namespace uses

### ISSUE THREAD

**Issue by MattKotsenas** [rjmurillo/moq.analyzers#56]
It's unclear if this is a bug in the analyzer, bug in the test harness or something else.

However, the namespace shouldn't impact the analyzer. So once our tests are in a good spot, we should investigate.

## Repro

```diff
"""
- namespace Moq.Analyzers.Test.DataAbstractClass.WithArgsPassed;
-
internal abstract class AbstractGenericClassWithCtor<T>
{
```

**Comment by rjmurillo** [comment]
Looks like this was fixed with #73 

There were two classes of issues clean up with that PR. One that cleaned up passing potentially null to places we shouldn't be, another is a set of null dereference issues. There are no guards and asserts to trigger when unexpected conditions are encountered

**Comment by MattKotsenas** [comment]
This bug is fixed now. Updated the title to remind me to add test coverage for this.
